### PR TITLE
add consistency check on PFRecHitQTestECALMultiThreshold.thresholds size

### PR DESCRIPTION
**Pull request by fwyzard**
Add a consistency check on  `PFRecHitQTestECALMultiThreshold`'s `thresholds` parameter's size.

Clean up RecoParticleFlow/PFClusterProducer with clang-tity.

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
On 6/9/17 11:23 AM, Andrea Bocci wrote:
> Clean up RecoParticleFlow/PFClusterProducer with clang-tity.
> 

Please clarify what were the commands/arguments to do cleanup.
Thank you.

**Comment by fwyzard** [comment]
They are written in the commit message.

Enabled checks:
    modernize-loop-convert
    modernize-make-unique
    modernize-pass-by-value
    modernize-redundant-void-arg
    modernize-replace-auto-ptr
    modernize-shrink-to-fit
    modernize-use-auto
    modernize-use-default
    modernize-use-nullptr
    modernize-use-override

Or do you want to know the exact commands ?

**Comment by slava77** [comment]
On 6/9/17 11:36 AM, Andrea Bocci wrote:
> They are written in the commit message.
> 
> Enabled checks:
> modernize-loop-convert
> modernize-make-unique
> modernize-pass-by-value
> modernize-redundant-void-arg
> modernize-replace-auto-ptr
> modernize-shrink-to-fit
> modernize-use-auto
> modernize-use-default
> modernize-use-nullptr
> modernize-use-override
> 
> Or do you want to know the exact commands ?
> 

I was looking for a copy-pasteable reference.

**Comment by fwyzard** [comment]
If you'd like instructions on how I'm running clang-tidy, I can share them in the hypernews.
I don;t think they belong in the commit messages or in the pull request itself (should I also include how I run an editor or how I compile or profile the code ?).

**Review by emanueledimarco — COMMENTED** [review]
Change is OK for me

File: RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc [hunk 1]
```diff
@@ -186,20 +183,20 @@ void PFCTRecHitProducer::produce(edm::Event& iEvent,
     //Loop on the calotower constituents and search for HCAL
     double dead = 0.;
     double alive = 0.;
-    for(unsigned int i=0;i< hits.size();++i) {
-      if(hits[i].det()==DetId::Hcal) { 
+    for(auto hit : hits) {
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc]: it seems OK to make a copy here. (```(auto hit : hits)``` operates on copies from the vector)
Was it a manual choice or is there some automated decision based on class content analysis?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/19180	bdc76cf
- implemented in line with the description. The functional change is in https://github.com/cms-sw/cmssw/pull/19180/commits/bdc76cfc872fd5f33432dad035f25ed541a0f46c?w=1 . The rest is clang-tity and related style improvements.
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by davidlange6** [comment]
+1

File: RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc (hunk 1 above)
Comment (fwyzard) [inline RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc]: That was the automatic conversion by 
```
clang-tidy -checks=modernize-loop-convert -fix RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc
```

**Pull request merged**